src/python/gem5/components/memory/memory.py

```diff
@@ -79,6 +79,8 @@ def __init__(
         interleaving_size: Union[int, str],
         size: Optional[str] = None,
         addr_mapping: Optional[str] = None,
+        holes: Optional[List[AddrRange]] = None,
+        # should holes be inserted here, or in the set_memory_range method?
```

Comment:
I've made this change as well, and have the x86 board just add the hole itself in the stdlib, so the user will be able to do more than 3 GB with the caveat that they are losing out on 1 GB because of the hole. So if they wanted 8GB of RAM they need to pass in a 9GB size.